From 07f2cd2724560b4d7870e553f2900bb7265183e1 Mon Sep 17 00:00:00 2001
From: Christian Krause <chkr@plauener.de>
Date: Wed, 26 Aug 2009 22:51:47 +0000
Subject: fix parallel build

1. adding some missing dependencies.

2. If the recursive make is called explicitely AND if the name for
the variable is SUBDIRS then make consider the target containing
the "make -C" call and the content of the SUBDIRS variable
as independent targets and it will execute them in parallel.
To prevent this behavior rename the special variable SUBDIRS to
SUBDIR.
---
diff --git a/extensions/Exporters/FlickrExport/Makefile.am b/extensions/Exporters/FlickrExport/Makefile.am
index 812904c..45e23fd 100644
--- a/extensions/Exporters/FlickrExport/Makefile.am
+++ b/extensions/Exporters/FlickrExport/Makefile.am
@@ -27,7 +27,7 @@ PKGS =					\
 	-pkg:gtk-sharp-2.0		\
 	-pkg:glade-sharp-2.0
 
-SUBDIRS = 			\
+SUBDIR = 			\
 	FlickrNet
 
 RESOURCES =				\
@@ -40,7 +40,7 @@ mpack: $(PLUGIN_ASSEMBLY)
 	mautil p $(PLUGIN_ASSEMBLY)
 
 $(PLUGIN_ASSEMBLY): $(PLUGIN_SOURCES) $(PLUGIN_MANIFEST)
-	$(MAKE) -C $(SUBDIRS)
+	$(MAKE) -C $(SUBDIR)
 	$(CSC_LIB) -out:$@ $(PLUGIN_SOURCES) $(REFS) $(PKGS) $(ASSEMBLIES) $(RESOURCES)
 
 plugindir = $(pkglibdir)/extensions
diff --git a/extensions/Exporters/PicasaWebExport/Makefile.am b/extensions/Exporters/PicasaWebExport/Makefile.am
index 45e3bac..e123ada 100644
--- a/extensions/Exporters/PicasaWebExport/Makefile.am
+++ b/extensions/Exporters/PicasaWebExport/Makefile.am
@@ -28,7 +28,7 @@ PKGS =					\
 	-pkg:gtk-sharp-2.0		\
 	-pkg:glade-sharp-2.0
 
-SUBDIRS = 			\
+SUBDIR = 			\
 	google-sharp
 
 RESOURCES =				\
@@ -41,7 +41,7 @@ mpack: $(PLUGIN_ASSEMBLY)
 	mautil p $(PLUGIN_ASSEMBLY)
 
 $(PLUGIN_ASSEMBLY): $(PLUGIN_SOURCES) $(PLUGIN_MANIFEST)
-	$(MAKE) -C $(SUBDIRS)
+	$(MAKE) -C $(SUBDIR)
 	$(CSC_LIB) -out:$@ $(PLUGIN_SOURCES) $(REFS) $(PKGS) $(ASSEMBLIES) $(RESOURCES)
 
 plugindir = $(pkglibdir)/extensions
diff --git a/extensions/Exporters/SmugMugExport/Makefile.am b/extensions/Exporters/SmugMugExport/Makefile.am
index 397461f..2f0dc2b 100644
--- a/extensions/Exporters/SmugMugExport/Makefile.am
+++ b/extensions/Exporters/SmugMugExport/Makefile.am
@@ -31,7 +31,7 @@ RESOURCES =				\
 	-resource:$(srcdir)/$(PLUGIN_MANIFEST)	\
 	-resource:$(srcdir)/$(PLUGIN_NAME).glade
 
-SUBDIRS =                      \
+SUBDIR =                      \
 	SmugMugNet
 
 
@@ -41,7 +41,7 @@ mpack: $(PLUGIN_ASSEMBLY)
 	mautil p $(PLUGIN_ASSEMBLY)
 
 $(PLUGIN_ASSEMBLY): $(PLUGIN_SOURCES) $(PLUGIN_MANIFEST)
-	$(MAKE) -C $(SUBDIRS)
+	$(MAKE) -C $(SUBDIR)
 	$(CSC_LIB) -out:$@ $(PLUGIN_SOURCES) $(REFS) $(PKGS) $(ASSEMBLIES) $(RESOURCES)
 
 plugindir = $(pkglibdir)/extensions
diff --git a/extensions/Exporters/TabbloExport/Makefile.am b/extensions/Exporters/TabbloExport/Makefile.am
index b7465d6..acf2d24 100644
--- a/extensions/Exporters/TabbloExport/Makefile.am
+++ b/extensions/Exporters/TabbloExport/Makefile.am
@@ -36,7 +36,7 @@ PKGS = \
 	-pkg:gtk-sharp-2.0 \
 	-pkg:glade-sharp-2.0 
 
-SUBDIRS = \
+SUBDIR = \
 	Tabblo
 
 RESOURCES = \
@@ -50,7 +50,7 @@ mpack: $(PLUGIN_ASSEMBLY)
 	mautil p $(PLUGIN_ASSEMBLY)
 
 $(PLUGIN_ASSEMBLY): $(PLUGIN_SOURCES) $(PLUGIN_MANIFEST)
-	$(MAKE) -C $(SUBDIRS)
+	$(MAKE) -C $(SUBDIR)
 	$(CSC_LIB) -out:$@ $(PLUGIN_SOURCES) $(REFS) $(PKGS) $(ASSEMBLIES) $(RESOURCES)
 
 plugindir = $(pkglibdir)/extensions
diff --git a/src/Makefile.am b/src/Makefile.am
index 02add7d..ba9bcaf 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -509,13 +509,13 @@ FSpot.JobScheduler.dll: $(JOBSCHEDULER_CSFILES) FSpot.Utils.dll
 
 FSpot.Bling.dll.mdb: FSpot.Bling.dll
 
-FSpot.Bling.dll: $(BLING_CSFILES)
+FSpot.Bling.dll: $(BLING_CSFILES) FSpot.Utils.dll
 	@echo -e "\n*** Compiling $@"
 	$(CSC_LIB) -out:$@ $(EXTRAFLAGS) $(BLING_CSFILES) $(BLING_ASSEMBLIES)
 
 FSpot.Widgets.dll.mdb: FSpot.Widgets.dll
 
-FSpot.Widgets.dll: $(WIDGETS_CSFILES) FSpot.Utils.dll FSpot.Core.dll
+FSpot.Widgets.dll: $(WIDGETS_CSFILES) FSpot.Utils.dll FSpot.Core.dll Cms.dll
 	@echo -e "\n*** Compiling $@"
 	$(CSC_LIB) -out:$@ $(EXTRAFLAGS) $(WIDGETS_CSFILES) $(WIDGETS_ASSEMBLIES)
 
--
cgit v0.8.2
